### PR TITLE
Microsoft ATP With Common API

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
reference: demisto/etc#20893

## Description
Implemented MicrosoftApiModule
Added support to authenticate using a self-deployed Azure application.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md [hunk 1]
```diff
@@ -13,14 +13,40 @@ Microsoft Defender Advanced Threat Protection Get Machine Action Status
...
-## Configure Microsoft Defender Advanced Threat Protection on Cortex XSOAR
+## Required Permissions
+##### Microsoft Graph Permissions:
+* Directory.Read.All - Delegated
+* ThreatIndicators.Read.All - Application
+* ThreatIndicators.ReadWrite.OwnedBy - Application
+##### WindowsDefenderATP Permissions:
+* AdvancedQuery.Read.All - Application
+* Alert.Read.All - Application
+* Alert.ReadWrite.All - Application
```
Comment (Itay4) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md:24]: this permission includes the previous permission, why do we need both?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md [hunk 2]
```diff
@@ -13,14 +13,40 @@ Microsoft Defender Advanced Threat Protection Get Machine Action Status
...
+* Event.Write - Application
+* File.Read.All - Application
+* Ip.Read.All - Application
+* Machine.CollectForensics - Application
+* Machine.Isolate - Application
+* Machine.Offboard - Application
+* Machine.Read.All - Application
```
Comment (Itay4) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md:31]: we have a few permissions here we ask for both application and delegated? are we sure we need these?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md (hunk 1 above)
Comment (Bargenish) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md:24]: We don't, I took it from our app registrations. I'll go through Microsoft API reference and split the permissions for each command.

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md (hunk 2 above)
Comment (Bargenish) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md:31]: I'll arrange it better with the  Microsoft API reference

**Comment by Bargenish** [comment]
@Itay4 I used the examples.txt file and ran manually most of the command.

**Comment by Itay4** [comment]
> @Itay4 I used the examples.txt file and ran manually most of the command.

please make sure to run all commands, either manually or with test playbook, in both setups: self deployed app and demisto app

**Comment by kirbles19** [comment]
@Bargenish Done.

**Pull request merged**